src/App.js

```diff
@@ -23,23 +21,26 @@ class App extends Component {
   };
 
   componentDidMount() {
-    // There is a lot improvement that you should here as well
-    //  make sure you catch errors that might happen
-    // an example is a network error, try and turn off internet and see what happens
+
+  try{
     axios
       .get("https://jsonplaceholder.typicode.com/todos")
-      .then(res => this.setState({ main: res.data }));
+      .then(res => this.setState(state =>({
+        main: res.data 
+      })));
+}catch(e){
```

Comment:
Did you know that the try and catch catches the exactly error that happened try and log the e

Reply:
Oh yeah
It's giving me INTERNET_DISCONNECTED and Network_Error errors in the console